### PR TITLE
drivers: lora: sx1276: Added support for RSSI and SNR in recv

### PR DESCRIPTION
**Pull request by KwonTae-young**
[LoRaMac-node](https://github.com/Lora-net/LoRaMac-node) can get `RSSI` and `SNR` value through [RxDone()](https://github.com/Lora-net/LoRaMac-node/blob/develop/src/mac/LoRaMac.c#L806-L820).
``` c
static void OnRadioRxDone( uint8_t *payload, uint16_t size, int16_t rssi, int8_t snr )
{
    RxDoneParams.LastRxDone = TimerGetCurrentTime( );
    RxDoneParams.Payload = payload;
    RxDoneParams.Size = size;
    RxDoneParams.Rssi = rssi;
    RxDoneParams.Snr = snr;

    LoRaMacRadioEvents.Events.RxDone = 1;

    if( ( MacCtx.MacCallbacks != NULL ) && ( MacCtx.MacCallbacks->MacProcessNotify != NULL ) )
    {
        MacCtx.MacCallbacks->MacProcessNotify( );
    }
}
```

This patch uses lora_recv() to get `RSSI` and `SNR` values.

**samples/drivers/lora/receive**
```
*** Booting Zephyr OS build zephyr-v1.13.0-13290-g890e9a30ebee  ***                                        
[00:00:00.222,000] <inf> sx1276: SX1276 Version:12 found                                                   
[00:00:01.086,000] <inf> lora_receive: Received data: helloworld (RSSI:-97dBm, SNR:7dBm)                   
[00:00:02.090,000] <inf> lora_receive: Received data: helloworld (RSSI:-100dBm, SNR:6dBm)                  
[00:00:03.094,000] <inf> lora_receive: Received data: helloworld (RSSI:-102dBm, SNR:5dBm)                  
[00:00:04.099,000] <inf> lora_receive: Received data: helloworld (RSSI:-97dBm, SNR:7dBm)                   
[00:00:05.103,000] <inf> lora_receive: Received data: helloworld (RSSI:-98dBm, SNR:8dBm)                   
[00:00:06.107,000] <inf> lora_receive: Received data: helloworld (RSSI:-98dBm, SNR:8dBm)                   
[00:00:07.111,000] <inf> lora_receive: Received data: helloworld (RSSI:-98dBm, SNR:8dBm)                   
[00:00:08.115,000] <inf> lora_receive: Received data: helloworld (RSSI:-98dBm, SNR:8dBm)                   
[00:00:09.119,000] <inf> lora_receive: Received data: helloworld (RSSI:-98dBm, SNR:8dBm)                   
[00:00:10.123,000] <inf> lora_receive: Received data: helloworld (RSSI:-98dBm, SNR:8dBm)                   
[00:00:11.127,000] <inf> lora_receive: Received data: helloworld (RSSI:-98dBm, SNR:8dBm)
```

### CODE REVIEW

File: drivers/lora/sx1276.c [hunk 1]
```diff
@@ -53,6 +53,8 @@ struct sx1276_data {
 	RadioEvents_t sx1276_event;
 	u8_t *rx_buf;
 	u8_t rx_len;
+	s16_t rx_rssi;
+	s8_t rx_snr;
```
Comment (tbursztyka) [inline drivers/lora/sx1276.c]: revert the order of the 2 attributes for the alignment to look obvious.

File: drivers/lora/sx1276.c [hunk 2]
```diff
@@ -371,6 +375,8 @@ static int sx1276_lora_recv(struct device *dev, u8_t *data, u8_t size,
 	 * wise method to fix this!
 	 */
 	memcpy(data, dev_data.rx_buf, dev_data.rx_len);
+	memcpy(rssi, &dev_data.rx_rssi, sizeof(dev_data.rx_rssi));
```
Comment (tbursztyka) [inline drivers/lora/sx1276.c]: memcpy() for things smaller than u32_t ? 

Assign directly. But check for NULL first.

**Review by tbursztyka — CHANGES_REQUESTED** [review]

File: drivers/lora/sx1276.c (hunk 1 above)
Comment (KwonTae-young) [reply drivers/lora/sx1276.c]: Done

File: drivers/lora/sx1276.c (hunk 2 above)
Comment (KwonTae-young) [reply drivers/lora/sx1276.c]: Done

File: drivers/lora/sx1276.c (hunk 1 above)
Comment (tbursztyka) [reply drivers/lora/sx1276.c]: no now you really broke the alignment. I meant to to so:

```
u8_t rx_len;
s8_t rx_snr;
s16_t rx_rssi;
```

Btw, rssi is always mesured on rx as well as snr, just call them rssi and snr, no need of the rx_ prefix.

File: drivers/lora/sx1276.c [hunk 3]
```diff
@@ -372,6 +376,12 @@ static int sx1276_lora_recv(struct device *dev, u8_t *data, u8_t size,
 	 */
 	memcpy(data, dev_data.rx_buf, dev_data.rx_len);
 
+	if (rssi != NULL)
```
Comment (tbursztyka) [inline drivers/lora/sx1276.c]: you need no add { } even on one line if statement, that's a style rule in Zephyr.

File: drivers/lora/sx1276.c [hunk 4]
```diff
@@ -372,6 +376,12 @@ static int sx1276_lora_recv(struct device *dev, u8_t *data, u8_t size,
...
+		*rssi = dev_data.rx_rssi;
+
+	if (snr != NULL)
```
Comment (tbursztyka) [inline drivers/lora/sx1276.c]: same here

**Review by tbursztyka — CHANGES_REQUESTED** [review]

**Review by Mani-Sadhasivam — APPROVED** [review]

File: drivers/lora/sx1276.c (hunk 3 above)
Comment (KwonTae-young) [reply drivers/lora/sx1276.c]: Done :sweat_smile:

**Review by tbursztyka — APPROVED** [review]
lgtm

**Pull request merged**